Marsiple/Common/Network/MartianApiManager.swift

```diff
@@ -54,16 +54,20 @@ private extension MartianApiManager {
         task.resume()
     }
     
-    static func postData<DataType: Codable>(data: DataType, url: URL) {
+    static func postData<DataType: Codable>(data: DataType, url: URL, success: @escaping (String) -> Void, failure: @escaping (String) -> Void) {
         guard let json = encodeData(data: data) else { return }
         let request = setupRequest(url: url, method: HTTPRequestMethod.post, body: json)
         let session = URLSession.shared
         let task = session.dataTask(with: request) { data, response, error in
             if let error = error {
```

Comment:
What happens when the error is not `nil` and we encountered an error? The code should stop executing and return, whilst notifying the caller using the provided `callbacks`.